### PR TITLE
Schema.org Plugins

### PR DESCRIPTION
**Pull request by shazmasiddiqui**
## plg_system_schema
- [x] Create a schema form in article edit view
- [x] Add 'Schema Type' list form field with 'None' selected by default

## plg_schemaorg_xxx
- [x] Create a new option in existing "Schema Type" list field in Schema form
- [x] Show form fields of selected schema type
- [x] Update type of form fields in schema form
- [x] Store form data in JSON format
- [x] Save into database
- [x] Fetch schema data from database and inject schema in head tag

### CODE REVIEW

File: administrator/components/com_schemaorg/src/Controller/DisplayController.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace ShazmaSiddiqui\Component\Schemaorg\Administrator\Controller;
```
Comment (ditsuke) [inline administrator/components/com_schemaorg/src/Controller/DisplayController.php]: pointer: component should be under the `Joomla\Component` namespace

File: administrator/components/com_schemaorg/schemaorg.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<extension type="component" method="upgrade">
+
+    <name>Schemaorg</name>
+    <creationDate>July 2022</creationDate>
+    <author>Shazma Siddiqui</author>
```
Comment (ditsuke) [inline administrator/components/com_schemaorg/schemaorg.xml]:
```suggestion
    <author>Joomla! Project</author>
```

File: administrator/components/com_schemaorg/src/Controller/DisplayController.php (hunk 1 above)
Comment (anuragteapot) [inline administrator/components/com_schemaorg/src/Controller/DisplayController.php]: namespace should be `joomla`

File: administrator/components/com_schemaorg/src/View/Schemaorg/HtmlView.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace ShazmaSiddiqui\Component\Schemaorg\Administrator\View\Schemaorg;
```
Comment (anuragteapot) [inline administrator/components/com_schemaorg/src/View/Schemaorg/HtmlView.php]: `namespace `

File: administrator/components/com_schemaorg/schemaorg.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    <version>0.0.1</version>
+    <description>
+        A Schema.org component
+    </description>
+
+    <namespace path="src/">ShazmaSiddiqui\Component\Schemaorg</namespace>
```
Comment (anuragteapot) [inline administrator/components/com_schemaorg/schemaorg.xml]: `namespace `

File: administrator/components/com_schemaorg/services/provider.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+use Joomla\CMS\Extension\Service\Provider\ComponentDispatcherFactory;
+use Joomla\CMS\Extension\Service\Provider\MVCFactory;
+use Joomla\CMS\MVC\Factory\MVCFactoryInterface;
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+
+return new class implements ServiceProviderInterface {
+    
+    public function register(Container $container): void {
+        $container->registerServiceProvider(new MVCFactory('\\ShazmaSiddiqui\\Component\\Schemaorg'));
```
Comment (anuragteapot) [inline administrator/components/com_schemaorg/services/provider.php]: `joomla`

**Comment by anuragteapot** [comment]
@shazmasiddiqui Please add here what things you added and how to test it

File: administrator/components/com_schemaorg/src/Controller/DisplayController.php (hunk 1 above)
Comment (shazmasiddiqui) [reply administrator/components/com_schemaorg/src/Controller/DisplayController.php]: done

Comment (shazmasiddiqui) [reply administrator/components/com_schemaorg/src/Controller/DisplayController.php]: done

File: plugins/system/schema/schema.php [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+        $event   = AbstractEvent::create(
+            'onSchemaPrepareForm',
+            [
+                'subject' => $form,
+            ]
+        );
+
+        PluginHelper::importPlugin('schemaorg');
+        $this->app->getDispatcher()->dispatch('onSchemaPrepareForm', $event);
```
Comment (bembelimen) [inline plugins/system/schema/schema.php:125]: This should be in the new event format.
Two example:
- https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/Workflow/WorkflowServiceTrait.php#L93-L104
- https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/Workflow/Workflow.php#L361-L396

**Pull request closed without merging**